BUILDING_CALICO.md

```diff
@@ -1,7 +1,7 @@
 # Building Calico components
 
 This document describes how to build a particular versioned Calico release from source. It describes how to
-build the following release components.
+build the following release components. Most developers should instead start with the [developer guide](DEVELOPER_GUIDE.md)
```

Comment:
What's the difference between these two guides? Can we explicitly call it out here?

From my understanding, this guide builds only from current **master** of each project, whereas the Developer Guide shows you how to use custom / personal branches instead of master?

Reply:
Yeah, I think we can actually ditch the old one. I'll make some changes here.